docs/Parameters.rst

```diff
@@ -538,6 +538,18 @@ Learning Control Parameters
 
       -  note that the parent output ``w_p`` itself has smoothing applied, unless it is the root node, so that the smoothing effect accumulates with the tree depth
 
+-  ``interaction_constraints`` :raw-html:`<a id="interaction_constraints" title="Permalink to this parameter" href="#interaction_constraints">&#x1F517;&#xFE0E;</a>`, default = ``""``, type = string
+
+   -  controls which features can appear in the same branch
+
+   -  for CLI, specify as lists separated by commas, e.g. ``[0,1,2],[2,3]``
+
+   -  for Python, specify as list of lists, e.g. ``[[0, 1, 2], [2, 3]]``
+
+   -  for R, specify as a string similar to the CLI format (ensuring there are no spaces), e.g. ``"[0,1,2],[2,3]"``
```

Comment:
For R, I think that we should make the interface "a list of character vectors, where each vector is a list of feature names", and then inside the R package turn the feature names into indices.

I think passing a string of an array in a comma-delimited list puts too much burden on R users.  @Laurae2 what do you think?

Reply:
@jameslamb Good suggestion, thanks! I've tried to implement this, but I'm not that familiar with R so would be great if you could take a look when you get a chance.

Reply:
I think it would be ok to say here that it is not supported in the R package yet, and then create an issue like `[R-package] add support for interaction_constraints` and add it to #2302 . Then I or another R maintainer can get to it without delaying this pull request.

Reply:
Done, see #3130